### PR TITLE
feat: add run casdoor through docker

### PR DESCRIPTION
**Pull request by oranges-eating**
Signed-off-by: killer <1533063601@qq.com>

### CODE REVIEW

**Comment by oranges-eating** [comment]
@hsluoyz If there is no problem in this part, I will do the docker installation of casnode.

**Comment by hsluoyz** [comment]
@Kininaru @ErikQQY plz review.

**Review by ErikQQY — APPROVED** [review]
I tested locally, everything works fine for me.

Screenshot from 2021-07-13 21-26-36

File: go-dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.17-rc-buster
+WORKDIR /casdoor
+COPY ./ /casdoor
+RUN go env -w CGO_ENABLED=0 GOPROXY=https://goproxy.io,direct GOOS=linux GOARCH=amd64 && apt update && apt install sudo && wget https://nodejs.org/dist/v12.22.0/node-v12.22.0-linux-x64.tar.gz && sudo tar xf node-v12.22.0-linux-x64.tar.gz && sudo apt install wait-for-it
+ENV PATH=$PATH:/casdoor/node-v12.22.0-linux-x64/bin
```
Comment (sh1luo) [inline go-dockerfile:5]: I think if there are multiple commands should be displayed in separate lines to increase readability, like this:
```shell
RUN commandA &&
         commandB &&
         commandC &&
         and etc...
```

File: go-dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+RUN npm install -g yarn && cd web && yarn install && yarn run build && rm -rf node_modules && cd /casdoor && go build main.go
+FROM alpine:3.7
+COPY --from=0 /casdoor   /
+COPY --from=0 /usr/bin/wait-for-it  /
+RUN set -eux && sed -i 's/dl-cdn.alpinelinux.org/mirrors.ustc.edu.cn/g' /etc/apk/repositories && apk update && apk upgrade && apk add bash
+CMD ./wait-for-it db:3306 && ./main
```
Comment (sh1luo) [inline go-dockerfile:11]: Same as above.

**Review by sh1luo — CHANGES_REQUESTED** [review]

**Comment by oranges-eating** [comment]
@hsluoyz Is there still any problem? If there is no problem, I will write casnode's dockerfile

**Comment by oranges-eating** [comment]
@hsluoyz Is have any question?

File: README.md [hunk 3]
```diff
@@ -47,6 +47,18 @@ Backend | RESTful API backend for Casdoor | Golang + Beego + MySQL | https://git
     git clone https://github.com/casbin/casdoor
     ```
 
+## RUN through Docker
```
Comment (hsluoyz) [inline README.md:50]: Run

File: README.md [hunk 4]
```diff
@@ -47,6 +47,18 @@ Backend | RESTful API backend for Casdoor | Golang + Beego + MySQL | https://git
     git clone https://github.com/casbin/casdoor
     ```
 
+## RUN through Docker
+- Install Docker and Docker-compose,you see [docker](https://docs.docker.com/get-docker/) and [docker-compose](https://docs.docker.com/compose/install/)
+- Vi casdoor/conf/app.conf
```
Comment (hsluoyz) [inline README.md:52]: vi ?

File: README.md [hunk 5]
```diff
@@ -47,6 +47,18 @@ Backend | RESTful API backend for Casdoor | Golang + Beego + MySQL | https://git
...
+- Install Docker and Docker-compose,you see [docker](https://docs.docker.com/get-docker/) and [docker-compose](https://docs.docker.com/compose/install/)
+- Vi casdoor/conf/app.conf
+- Modify dataSourceName = root:123@tcp(localhost:3306)/ to dataSourceName = root:123@tcp(db:3306)/
+- Execute the following command
+  ```shell
+  docker-compose up
+  ```
+- Open browser:
+
+  http://your-ip:8000/
```
Comment (hsluoyz) [inline README.md:60]: Does localhost work?

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    restart: always
+    image: mysql:8.0.25
+    ports:
+      - 3306:3306
+    environment:
+      MYSQL_ROOT_PASSWORD: 123
+    volumes:
+      - /usr/local/docker/mysql:/var/lib/mysql
+
+
```
Comment (hsluoyz) [inline docker-compose.yml:21]: No so many new lines

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by oranges-eating** [comment]
@hsluoyz already solved

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@oranges-eating put the docs to: https://casdoor.org/

**Pull request merged**